### PR TITLE
Bosh2 manifests

### PR DESCRIPTION
**Pull request by rkoster**
This PR adds bosh2 style manifests.
I have initially only added ops files for bosh and cf exporters.
A manifest can be generated by running:
```
bosh int manifests/prometheus.yml \
  -o manifests/operators/prometheus-bosh-exporter.yml \
  -o manifests/operators/prometheus-cf-exporter.yml 
```

### CODE REVIEW

File: manifests/prometheus.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  vm_type: default
+  stemcell: default
+  networks:
+  - name: default
+  jobs:
+    - name: nginx
+      release: prometheus
+      properties:
+        nginx:
+          grafana:
```
Comment (frodenas) [inline manifests/prometheus.yml]: We have 2 levels of authentication for grafana. Can we delete the `grafana` section from the `nginx` job please? Grafana will provide its own authentication method (that might be configured to use UAA for example).

File: manifests/operators/prometheus-bosh-exporter.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        metrics:
+          environment: ((metrics_environment))
+        bosh:
+          url: ((bosh_url))
+          username: ((bosh_username))
+          password: ((bosh_password))
+          ca_cert: ((bosh_ca_cert)
+
+# Grafana Dashboards
+- type: replace
```
Comment (frodenas) [inline manifests/operators/prometheus-bosh-exporter.yml]: These dashboards are only useful if you have deployed the `node-exporter` addon, so better to add them on a `node-exporter` operator.

File: manifests/operators/prometheus-bosh-exporter.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  path: /instance_groups/name=grafana/jobs/-
+  value: 
+    name: system_dashboards
+    release: prometheus
+
+- type: replace
```
Comment (frodenas) [inline manifests/operators/prometheus-bosh-exporter.yml]: Same as above.

File: manifests/operators/prometheus-bosh-exporter.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  path: /instance_groups/name=prometheus/jobs/-
+  value: 
+    name: bosh_alerts
+    release: prometheus
+
+- type: replace
+  path: /instance_groups/name=prometheus/jobs/name=prometheus/properties/prometheus/rule_files/-
+  value: /var/vcap/packages/bosh_alerts/*.alerts
+
+# Alertmanager Grouping
```
Comment (frodenas) [inline manifests/operators/prometheus-bosh-exporter.yml]: You don't really need this section, unless you define an explicit receiver (I've never seen this use case).

File: manifests/operators/prometheus-cf-exporter.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    name: cf_exporter
+    release: prometheus
+    properties:
+      cf_exporter:
+        metrics:
+          environment: ((metrics_environment))
+        skip_ssl_verify: ((skip_ssl_verify))
+        cf:
+          api_url: ((cf_api_url))
+          username: ((cf_admin_username))
```
Comment (frodenas) [inline manifests/operators/prometheus-cf-exporter.yml]: Better to use uaa client id and secret, when using user & password, the cf exporter stop working after some time (when the refresh token expires).

File: manifests/operators/prometheus-remote-bosh-exporter.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+# Depends on ./prometheus-bosh-exporter.yml
```
Comment (frodenas) [inline manifests/operators/prometheus-remote-bosh-exporter.yml]: What's the goal of this remote exporter?

Comment (rkoster) [inline manifests/operators/prometheus-remote-bosh-exporter.yml]: I'm using it with bucc: https://github.com/starkandwayne/bucc/pull/20. If rolling out clean envs there is a circular dependency between the bosh tsdb health monitor endpoint and the prometheus deployment which you would deploy with said bosh. The solution is colocating the boshtsdb-exporter with bosh during initial deploy. And let prometheus scrap it once its deployed.

File: manifests/operators/prometheus-bosh-exporter.yml (hunk 4 above)
Comment (rkoster) [inline manifests/operators/prometheus-bosh-exporter.yml]: This allowed bosh-exporter and boshtsdb-exporter alerts to be grouped. As can be seen in the following payload (from alertmanager webhook) https://gist.github.com/rkoster/80509fc149a88d9ab350cf2b2d5092b7#file-grouped_alerts-json-L126-L129

File: manifests/operators/prometheus-cf-exporter.yml (hunk 5 above)
Comment (rkoster) [inline manifests/operators/prometheus-cf-exporter.yml]: which authorities does the client need?

Comment (frodenas) [inline manifests/operators/prometheus-cf-exporter.yml]: `cloud_controller.admin` (see https://github.com/cloudfoundry-community/cf_exporter#uaa-client)

File: manifests/operators/prometheus-bosh-exporter.yml (hunk 4 above)
Comment (frodenas) [inline manifests/operators/prometheus-bosh-exporter.yml]: Don't use the `bosh tsdb exporter` yet, it's useless right now, there aren't any dashboards not alerts.

File: manifests/operators/prometheus-remote-bosh-exporter.yml (hunk 6 above)
Comment (frodenas) [inline manifests/operators/prometheus-remote-bosh-exporter.yml]: Don't use the `bosh tsdb exporter` yet (see previous comment).

File: manifests/operators/prometheus-bosh-exporter.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    release: prometheus
+    properties:
+      bosh_exporter:
+        metrics:
+          environment: ((metrics_environment))
+        bosh:
+          url: ((bosh_url))
+          username: ((bosh_username))
+          password: ((bosh_password))
+          ca_cert: ((bosh_ca_cert)
```
Comment (rkoster) [inline manifests/operators/prometheus-bosh-exporter.yml]: missing `)`

File: manifests/operators/prometheus-bosh-exporter.yml (hunk 4 above)
Comment (rkoster) [inline manifests/operators/prometheus-bosh-exporter.yml]: fixed in: https://github.com/cloudfoundry-community/prometheus-boshrelease/pull/71/commits/1520b3aa638e48eab042de1facdcfd16f9e154bb

**Pull request merged**